File: solutions/lua/resistor-color-duo/1/resistor-color-duo.lua
return {
  value = function(colors)
    local bands = {
      black = 0,
      brown = 1,
      red = 2,
      orange = 3,
      yellow = 4,
      green = 5,
      blue = 6,
      violet = 7,
      grey = 8,
      white = 9
    }
    local bandOne = bands[colors[1]]
    local bandTwo = bands[colors[2]]
    return (bandOne * 10) + bandTwo
  end
}

File: solutions/elixir/resistor-color-duo/3/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value([first_band, second_band | _rest]) do
    get_resistance(first_band) * 10 + get_resistance(second_band)
  end

  defp get_resistance(color) do
    cond do
      color == :black -> 0
      color == :brown -> 1
      color == :red -> 2
      color == :orange -> 3
      color == :yellow -> 4
      color == :green -> 5
      color == :blue -> 6
      color == :violet -> 7
      color == :grey -> 8
      color == :white -> 9
      true -> 0
    end
  end
end
